src/snippets/snippets.ts

```diff
@@ -1,22 +1,47 @@
 import { Options } from "./options";
 
 export class RawSnippet {
-	trigger: string;
+	trigger: string | RegExp;
 	replacement: string;
 	options: string;
 	priority?: number;
 	description?: string;
+	flags?: string;
 }
 
 export class ParsedSnippet {
 	trigger: string;
 	replacement: string;
 	options: Options;
+	flags: string;
 	priority?: number;
 	description?: string;
-
+	
 	constructor(raw: RawSnippet) {
-		const parsed = {...raw, options: Options.fromSource(raw.options)};
+		// normalize triggers as strings
+		const override: RawSnippet = { ...raw }
+		if (raw.trigger instanceof RegExp) {
+			override.options = `r${raw.options}`;
+			override.trigger = raw.trigger.source;
+			override.flags = raw.trigger.flags;
+		}
+
+		// merge flags
+		const validFlags = [
+			// "d", // doesn't affect the search
+			// "g", // doesn't affect the pattern match and is almost certainly undesired behavior
+			"i",
+			"m",
+			"s",
+			"u",
+			"v",
+			// "y", // almost certainly undesired behavior
+		];
+		const resolvedFlags = Array.from(new Set(`${override.flags ?? ""}${raw.flags ?? ""}`.split("")))
```

Comment:
Is it necessary to include `${raw.flags ?? ""}` when `override` already contains all the flags from `raw` (because `override` is defined as `{...raw}`)?

Reply:
`override.flags` was clobbering that of `raw` in the case that trigger was a regexp. i've rewritten the related code in the constructor to make the values easier to trace and remove the use of `raw` outside of the initial construction of `override`.